src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -1236,33 +1249,43 @@ int publish_commit(rgw::sal::Object* obj,
           return ret;
         }
         // now try to make a bigger one
-	buffer::list obl;
-        int rval;
-        cls_2pc_queue_reserve(op, bl.length(), 1, &obl, &rval);
-        ret = rgw_rados_operate(
-	  dpp, res.store->getRados()->get_notif_pool_ctx(),
-          queue_name, &op, res.yield, librados::OPERATION_RETURNVEC);
+	// buffer::list obl;
+  //       int rval;
+  //       cls_2pc_queue_reserve(op, bl.length(), 1, &obl, &rval);
+  //       ret = rgw_rados_operate(
+	//   dpp, res.store->getRados()->get_notif_pool_ctx(),
+  //         queue_name, &op, res.yield, librados::OPERATION_RETURNVEC);
+        ret = rgw::redisqueue::reserve(conn, queue_name, bl.length(), res.yield);
         if (ret < 0) {
           ldpp_dout(dpp, 1) << "ERROR: failed to reserve extra space on queue: "
 			    << queue_name
 			    << ". error: " << ret << dendl;
           return (ret == -ENOSPC) ? -ERR_RATE_LIMITED : ret;
         }
-        ret = cls_2pc_queue_reserve_result(obl, topic.res_id);
-        if (ret < 0) {
-          ldpp_dout(dpp, 1) << "ERROR: failed to parse reservation id for "
-	    "extra space. error: " << ret << dendl;
-          return ret;
-        }
+      //   ret = cls_2pc_queue_reserve_result(obl, topic.res_id);
+      //   if (ret < 0) {
+      //     ldpp_dout(dpp, 1) << "ERROR: failed to parse reservation id for "
+	    // "extra space. error: " << ret << dendl;
+      //     return ret;
+      //   }
       }
-      std::vector<buffer::list> bl_data_vec{std::move(bl)};
+      // std::vector<buffer::list> bl_data_vec{std::move(bl)};
       librados::ObjectWriteOperation op;
-      cls_2pc_queue_commit(op, bl_data_vec, topic.res_id);
+
+      // FIXME: Is this commiting multiple messages over a single reservation?
+      // How to convert a bufferlist to a string or a vector of strings incase it is multiple messages?
+      // cls_2pc_queue_commit(op, bl_data_vec, topic.res_id);
+
+      // This needs to be fetched from bufferlist
+      std::string data = "Data to commit";
+
+      auto ret = rgw::redisqueue::commit(conn, queue_name, data, res.yield);
+
       aio_completion_ptr completion {librados::Rados::aio_create_completion()};
       auto pcc_arg = make_unique<PublishCommitCompleteArg>(queue_name, dpp);
       completion->set_complete_callback(pcc_arg.get(), publish_commit_completion);
       auto &io_ctx = res.store->getRados()->get_notif_pool_ctx();
```

Comment:
there will be no completion from RADOS here (because there is no RADOS operation being called).

Reply:
Do I simply comment out this part? 
I think I confused this part of the code to be operating on topic metadata part.